arangod/Aql/HashedCollectExecutor.cpp

```diff
@@ -355,9 +391,34 @@ HashedCollectExecutor::findOrEmplaceGroup(InputAqlItemRow& input) {
   ResourceUsageScope guard(_infos.getResourceMonitor(),
                            memoryUsageForGroup(_nextGroup, true));
 
+  std::unique_ptr<velocypack::Builder> builder;
+
+  if (_infos.getCollectRegister().value() != RegisterId::maxRegisterId) {
+    builder = std::make_unique<velocypack::Builder>();
+    builder->openArray();
+    if (_infos.getExpressionVariable() != nullptr) {
+      input.getValue(_infos.getExpressionRegister())
+          .toVelocyPack(_infos.getVPackOptions(), *builder,
+                        /*resolveExternals*/ false,
+                        /*allowUnindexed*/ false);
+    } else {
+      // copy variables / keep variables into result register
+      builder->openObject();
+      for (auto const& pair : _infos.getInputVariables()) {
+        builder->add(VPackValue(pair.first));
+        input.getValue(pair.second)
+            .toVelocyPack(_infos.getVPackOptions(), *builder,
+                          /*resolveExternals*/ false,
+                          /*allowUnindexed*/ false);
+      }
+      builder->close();
+    }
```

Comment:
Duplication of lines 338-354. Can we extract this into a small helper function?

Reply:
Done